src/net_processing.cpp

```diff
@@ -3375,6 +3515,34 @@ void PeerManagerImpl::ProcessCompactBlockTxns(CNode& pfrom, Peer& peer, const Bl
     return;
 }
 
+void PeerManagerImpl::PushPrivateBroadcastTx(CNode& node)
+{
+    Assume(node.IsPrivateBroadcastConn());
+
+    auto opt_tx = m_tx_for_private_broadcast.GetTxForBroadcast();
+    if (!opt_tx) {
+        LogDebug(BCLog::PRIVATE_BROADCAST,
+                 "Disconnecting: no more transactions for private broadcast (connected in vain), peer=%d%s",
+                 node.GetId(),
+                 node.LogIP(fLogIPs));
+        node.fDisconnect = true;
+        return;
+    }
+    const CTransactionRef& tx{*opt_tx};
```

Comment:
Reaching here can be quite delayed from when the tx was actually scheduled. Could we check here that the tx is still acceptable to our mempool before pushing?

Reply:
Looks a bit of an overkill to me. It could be delayed but normally it is not and the worst that can happen is that this will be detected later during the rebroadcast attempt. No strong opinion but I somewhat lean on leaving it as it is.